src/snowflake/snowpark/mock/_connection.py

```diff
@@ -323,6 +323,10 @@ def __init__(self, options: Optional[Dict[str, Any]] = None) -> None:
             self._oob_telemetry.disable()
         else:
             self._oob_telemetry.log_session_creation(self._connection_uuid)
+        # thread safe param protection
+        self._thread_safe_session_enabled = self._options.get(
```

Comment:
is that needed here? or should always be false for the mock?

Reply:
agreed

Reply:
We will read connection parameter given by user. If not, we will disable by default.

Reply:
i recall in the doc that supporting thread safe for local testing is out of scope at this moment, so i don't expect anyone can turn this on for mock. if that is the case, then i think this flag should always just be false, even if "PYTHON_SNOWPARK_ENABLE_THREAD_SAFE_SESSION" is set to true. is there any code under mock that is currently using this parameter?